presto-hive/src/main/java/com/facebook/presto/hive/ParquetRecordWriterUtil.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.presto.hive.RecordFileWriter.ExtendedRecordWriter;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.ql.exec.FileSinkOperator.RecordWriter;
+import org.apache.hadoop.hive.ql.io.HiveOutputFormat;
+import org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat;
+import org.apache.hadoop.hive.ql.io.parquet.write.ParquetRecordWriterWrapper;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapred.JobConf;
+import org.apache.hadoop.mapred.Reporter;
+import parquet.hadoop.ParquetFileWriter;
+import parquet.hadoop.ParquetRecordWriter;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.Properties;
+
+public final class ParquetRecordWriterUtil
+{
+    private static final Field REAL_WRITER_FIELD;
+    private static final Field INTERNAL_WRITER_FIELD;
+    private static final Field FILE_WRITER_FIELD;
+
+    static {
+        try {
+            REAL_WRITER_FIELD = ParquetRecordWriterWrapper.class.getDeclaredField("realWriter");
+            INTERNAL_WRITER_FIELD = ParquetRecordWriter.class.getDeclaredField("internalWriter");
+            FILE_WRITER_FIELD = Class.forName(
+                    "parquet.hadoop.InternalParquetRecordWriter",
+                    false,
+                    ParquetRecordWriter.class.getClassLoader())
+                    .getDeclaredField("parquetFileWriter");
+
+            REAL_WRITER_FIELD.setAccessible(true);
+            INTERNAL_WRITER_FIELD.setAccessible(true);
+            FILE_WRITER_FIELD.setAccessible(true);
+        }
+        catch (ReflectiveOperationException e) {
+            throw new AssertionError(e);
+        }
+    }
+
+    private ParquetRecordWriterUtil() {}
+
+    public static RecordWriter createParquetWriter(Path target, JobConf conf, Properties properties, boolean compress)
+            throws IOException, ReflectiveOperationException
+    {
+        Object outputFormat = MapredParquetOutputFormat.class.getConstructor().newInstance();
```

Comment:
Why not just `MapredParquetOutputFormat outputFormat = new MapredParquetOutputFormat()`?  Looks like the method is public.